cli/src/cmd/forge/script/cmd.rs

```diff
@@ -0,0 +1,193 @@
+                    decoder = self.decode_traces(&script_config, &mut result, &known_contracts)?;
+                } else {
+                    // prepend predeploy libraries
+                    let mut lib_deploy = self.create_deploy_transactions(
+                        script_config.evm_opts.sender,
+                        script_config.sender_nonce,
+                        &predeploy_libraries,
+                    );
+
+                    if let Some(txs) = &mut result.transactions {
+                        for tx in txs.iter() {
+                            lib_deploy.push_back(TypedTransaction::Legacy(tx.clone().into()));
+                        }
+                        *txs = lib_deploy;
+                    }
+                }
+
+                self.show_traces(&script_config, &decoder, &mut result)?;
+
+                self.handle_broadcastable_transactions(
+                    &target,
+                    result.transactions,
+                    &mut decoder,
+                    &script_config,
+                )
+                .await?;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Reruns the execution with a new sender and relinks the libraries accordingly
+    async fn rerun_with_new_deployer(
+        &self,
+        project: Project,
+        script_config: &mut ScriptConfig,
+        new_sender: Address,
+        first_run_result: &mut ScriptResult,
+        default_known_contracts: BTreeMap<ArtifactId, CompactContractBytecode>,
+    ) -> eyre::Result<BTreeMap<ArtifactId, (Abi, Vec<u8>)>> {
+        // if we had a new sender that requires relinking, we need to
+        // get the nonce mainnet for accurate addresses for predeploy libs
+        let nonce = foundry_utils::next_nonce(
+            new_sender,
+            script_config
+                .evm_opts
+                .fork_url
+                .as_ref()
+                .expect("You must provide an RPC URL (see --fork-url) when broadcasting."),
+            None,
+        )
+        .await?;
+        script_config.sender_nonce = nonce;
+
+        let BuildOutput { contract, highlevel_known_contracts, predeploy_libraries, .. } = self
+            .link(
+                project,
+                default_known_contracts,
+                script_config.config.parsed_libraries()?,
+                new_sender,
+                nonce,
+            )?;
+
+        first_run_result.transactions =
+            Some(self.create_deploy_transactions(new_sender, nonce, &predeploy_libraries));
+
+        let result =
+            self.execute(script_config, contract, new_sender, &predeploy_libraries).await?;
+
+        first_run_result.success &= result.success;
+        first_run_result.gas = result.gas;
+        first_run_result.logs = result.logs;
+        first_run_result.traces.extend(result.traces);
+        first_run_result.debug = result.debug;
+        first_run_result.labeled_addresses.extend(result.labeled_addresses);
```

Comment:
<s>will there be duplicate labeled addresses when relinking?</s> (nope because labeled_addresses is a btreemap

Reply:
not duplicated, but because of duplicate traces, we were labeling the library address with the wrong label (fixed https://github.com/foundry-rs/foundry/pull/1766)